### PR TITLE
[feat]: [CDS-47351]: ArtifactoryRegistryServiceImplTest

### PR DESCRIPTION
**Pull request by vtxorxwitty**
## Describe your changes
This PR contains UT coverage improvements for ArtifactoryRegistryServiceImpl class.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40697)

### CODE REVIEW

**Review by satyam-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40697)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by satyam-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40697#-:-NHxiXhw5aXc8JpBTFB2:bnfp4nl)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by inderschera — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40697)* status: 2 file(s) to review. 1 of 2 approvals obtained. 

___
*[`960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java` line 59 at r1](https://reviewable.io/reviews/harness/harness-core/40697#-NHxk7jR5y_KtxYzJqVV:-NHxk7jR5y_KtxYzJqVW:b8w7g4a) ([raw file](https://github.com/harness/harness-core/blob/94d5a4acce9f0ef4f6ecf4af7fbb0362fcd052e4/960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java#L59)):*
> ```Java
>       Pattern.compile(tagRegex);
>     } catch (PatternSyntaxException e) {
>       throw new HintException("Please check tagRegex field in Artifactory artifact configuration.");
> ```

why are we removing hintWithExplanation exception?

**Review by vivekkarnatiharness — DISMISSED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40697)* status: :shipit: complete! ✓

**Review by vtxorxwitty — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40697)* status: 1 file(s) to review. 0 of 2 approvals obtained. 

___
*[`960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java` line 59 at r1](https://reviewable.io/reviews/harness/harness-core/40697#-NHxk7jR5y_KtxYzJqVV:-NHxvFTB3gu9_52SMzy4:bvljs2o) ([raw file](https://github.com/harness/harness-core/blob/94d5a4acce9f0ef4f6ecf4af7fbb0362fcd052e4/960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java#L59)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

why are we removing hintWithExplanation exception?
</blockquote></details>

Addressed.

File: 960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public BuildDetailsInternal getLastSuccessfulBuildFromRegex(ArtifactoryConfigReq
     } catch (PatternSyntaxException e) {
       throw NestedExceptionUtils.hintWithExplanationException(
           "Please check tagRegex field in Artifactory artifact configuration.",
-          String.format("TagRegex field contains an invalid regex value '%s'.", tagRegex),
+          String.format("TagRegex field contains an invalid regex value " + tagRegex),
```
[the remark this answers is not in the document]
Comment (vtxorxwitty) [reply 960-api-services/src/main/java/io/harness/artifactory/service/ArtifactoryRegistryServiceImpl.java:60]: Getting this error of format conversion, if not changed.. 

`Conversion = 's'
java.util.UnknownFormatConversionException: 
	at java.base/java.util.Formatter.checkText(Formatter.java:2732)
	at java.base/java.util.Formatter.parse(Formatter.java:2718)
	at java.base/java.util.Formatter.format(Formatter.java:2655)
	at java.base/java.util.Formatter.format(Formatter.java:2609)
	at java.base/java.lang.String.format(String.java:2897)
	at java.base/java.util.UnknownFormatConversionException.getMessage(UnknownFormatConversionException.java:66)
	at java.base/java.lang.Throwable.getLocalizedMessage(Throwable.java:396)
	at java.base/java.lang.Throwable.toString(Throwable.java:485)
	at java.base/java.lang.String.valueOf(String.java:2951)
	at java.base/java.io.PrintWriter.println(PrintWriter.java:837)
	at java.base/java.lang.Throwable$WrappedPrintWriter.println(Throwable.java:768)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:659)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:725)
	at org.assertj.core.util.Throwables.getStackTrace(Throwables.java:169)
	at org.assertj.core.error.ShouldBeInstance.<init>(ShouldBeInstance.java:67)
	at org.assertj.core.error.ShouldBeInstance.shouldBeInstance(ShouldBeInstance.java:34)
	at org.assertj.core.internal.Objects.assertIsInstanceOf(Objects.java:137)
	at org.assertj.core.api.AbstractAssert.isInstanceOf(AbstractAssert.java:414)
`

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40697)* status: 2 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by puthrayaharness — APPROVED** [review]

**Pull request merged**